### DISCUSSION TITLE
Force OTA Update on IKEA Devices

### DISCUSSION THREAD

**Discussion by djusHa** [Koenkk/zigbee2mqtt#5540] [Devices]
Hi there,
any possibility to force OTA Update on Ikea devices?

After resetting devices ( some light bulbs and remotes ) i can't control Light Scenes on Bulbs with the Remotes ( know issue ).

**Comment by Koenkk** [comment]
Check: https://www.zigbee2mqtt.io/information/ota_updates.html#update-to-latest-firmware

**Reply by djusHa** [reply]
Thanks, but it doesn't work.

There is already the latest Firmware on this Bulb.

How can i force it?

`INFO: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"id":"whz_decke_lampe"},"error":"Update of 'whz_decke_lampe' failed (No new image available)","status":"error","transaction":"lh59q-3"}`

`ERROR: Update of 'whz_decke_lampe' failed (No new image available)`

**Reply by Koenkk** [reply]
It's already on the latest firmware version, why update it again?

**Reply by djusHa** [reply]
To get Light Scenes working again.

**Reply by Koenkk** [reply]
This is not something we want to support I think. IMHO it makes no sense to update to the same firmware. 

However if you want this you can easily disable the check by removing this line https://github.com/Koenkk/zigbee-herdsman-converters/blob/d1cbb168e12345d66f581ef74d6884c713984182/lib/ota/common.js#L412 .

See https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html on how to modify `lib/ota/common.js` (the `lib` folder is next to the `devices.js` file mentioned in that tutorial)

**Reply by djusHa** [reply]
You right, it makes no sense to Update to the same Version, but there is no other Way to fix the Light Scenes, right?

Or did i miss something?

**Reply by Koenkk** [reply]
I don't know about that issue

**Comment by djusHa** [comment]
This one #1232

**Reply by Koenkk** [reply]
Ah that one, I've already spend a lot of time on fixing it but failed to do. I believe @MattWestb contacted Ikea about this. I think we can only fix this after Ikea tells us how it works :smile:

**Reply by djusHa** [reply]
I think it's like that :laughing: 

No way to force the Update, changed src, but the Bulb refuses the Update itself ...

**Reply by MattWestb** [reply]
I still think its working inZ2M if adding the scenes OK in the lights but its depends if @sjorge have getting all things in place with adding scenes.
IKEA is only having 3 "normal" stored scenes that is restored then the light is updated so it should not being necessarily to downgrading the firmware than all scenes commands is implemented and working its only making new scenes as you like and its working :-)

Edit: deCONZ and ZHA can forcing firmware update (thru local files) and downgrading the device firmware.

**Reply by sjorge** [reply]
Are the scene_id's fixed? If so, I think it would indeed just be adding the scenes.

I need to add hue/saturation support to scene_add, but ikea is not using those, so it should be fine with the current implementation.

**Reply by MattWestb** [reply]
From the sniffing i have dont is IKEA only setting up there scenes with xy but its for 2 months ago (they have changing the mode to scenes in there GW after that).
Adding the "magic groupe" to the the light and then adding scenes to it with the group ID and all the scene ID you can / like.
